### ISSUE TITLE
Handling single expression lambdas with `void` return type

### ISSUE THREAD

**Issue by msridhar** [uber/NullAway#36]
It looks like even with a single-expression lambda has a `void` return type, we are still checking the nullability of the single expression as if it were getting returned.  Don't have a small test case yet, will work on it.  Just want to record here so we don't forget

**Comment by msridhar** [comment]
Added a test case on the `ms-void-nullable-test-case` branch.  In short, this code gives a NullAway error when it shouldn't:
```java
public class NullAwayJava8NegativeCases {
  @FunctionalInterface
  interface RetNullableFunction {
    @Nullable
    Object getVal();
  }
  @FunctionalInterface
  interface VoidFunction {
    void doSomething();
  }
  static void wrapDoSomething() {
    RetNullableFunction r = () -> null;
    VoidFunction v = () -> r.getVal();
    v.doSomething();
  } 
}
```

**Comment by lazaroclapp** [comment]
This is really weird, because even if we are assuming `v.doSomething();` returns `null`, instead of `void`, we are still not using that `null` value anywhere and this should not produce an error, right? It is probably easy enough to add a check for `void` in the signature of the FI to our code here, but I am going to try to understand this a little better before that. This is not currently causing problems on any real codebase, right? Will take a look today if I can, or else later this week.
